src/Microsoft.DotNet.Build.CloudTestTasks/RunnerScripts/xunitrunner-perf/csvjsonconvertor.py

```diff
@@ -11,6 +11,42 @@
 
 log = helix.logs.get_logger()
 
+# read a row of csv data into dict
+def add_row(test, value, csvdict):
+    """
+        test names are of the form
+        System.ComponentModel.Tests.Perf_TypeDescriptorTests.GetConverter(typeToConvert: typeof(bool), expectedConverter: typeof(System.ComponentModel.BooleanConverter))
+        where the string before the first '(' are the identifiers for the test and the string after is function metadata
+        strip off the metadata when generating the namespace/function identifiers and combine them when again for function identifier
+        eg.: System.ComponentModel.Tests.Perf_TypeDescriptorTests.GetConverter(typeToConvert: typeof(bool), expectedConverter: typeof(System.ComponentModel.BooleanConverter))
+        will be converted to following identifiers
+        [System, ComponentModel, Tests, Perf_TypeDescriptorTests, GetConverter(typeToConvert: typeof(bool), expectedConverter: typeof(System.ComponentModel.BooleanConverter))]
+    """
+
+    funcMeta = test.split('(')
+    funcMeta = funcMeta[1:]
+    test = test.split('(')[0]
+    identifiers = test.split('.')
+    funcName = identifiers[len(identifiers) - 1]
+    for data in funcMeta:
+        funcName = funcName + '(' + data
+
+    identifiers[len(identifiers) - 1] = funcName
+    i=0
+    currdict = csvdict
+    while (i<len(identifiers) -1):
```

Comment:
This is just "for part in identifiers[:-2]". It's a lot clearer that way (and I had to delete my other comment, because I mis understood this loop without digging deeply into it).